### PR TITLE
--remove-orphans is ignored when using up --no-start

### LINKED ISSUE #5547: --remove-orphans is ignored when using up --no-start
Given the following configuration:

```yaml
# cat two_containers.yml
version: '2.1'
services:
  foo:
    image: my-web-service
  bar:
    image: my-web-service
# cat one_container.yml
version: '2.1'
services:
  foo:
    image: my-web-service
```

`--remove-orphans` removes the `bar` container when I do `up -d`:

```bash
# /apps/research/tools/docker-compose/1.17.1_no_cert_warnings/bin/docker-compose -f two_containers.yml up -d
Creating mytest_bar_1 ...
Creating mytest_foo_1 ...
Creating mytest_foo_1
Creating mytest_bar_1 ... done
# /apps/research/tools/docker-compose/1.17.1_no_cert_warnings/bin/docker-compose -f one_container.yml up -d --remove-orphans
Removing orphan container "mytest_bar_1"
mytest_foo_1 is up-to-date
```

However, `--remove-orphans` seems to be ignored with `up --no-start`:

```bash
# docker-compose -f two_containers.yml up --no-start
Creating network "mytest_default" with the default driver
Creating mytest_bar_1 ...
Creating mytest_foo_1 ...
Creating mytest_bar_1
Creating mytest_bar_1 ... done
# docker-compose -f one_container.yml up --no-start --remove-orphans
```

### PR DESCRIPTION
**Pull request by collin5**
Resolves #5547

### CODE REVIEW

**Comment by collin5** [comment]
Updated. Thank you!

File: compose/project.py [hunk 1]
```diff
@@ -638,7 +638,10 @@ def _find():
         if remove_orphans:
             for ctnr in orphans:
                 log.info('Removing orphan container "{0}"'.format(ctnr.name))
-                ctnr.kill()
+                try:
+                    ctnr.kill()
+                except Exception:
```
Comment (shin-) [inline compose/project.py]: We should catch `docker.errors.APIError` here as to not hide other legitimate exceptions.

**Comment by collin5** [comment]
Fixed! Thanks

**Comment by ulyssessouza** [comment]
Thank you @collin5 !
Could you please, squash the 3 commits?

**Comment by collin5** [comment]
ping @ulyssessouza

**Comment by chfmunich** [comment]
there is a browser addon i use exactly for this
https://github.com/sniklaus/youtube-watchmarker
https://addons.mozilla.org/firefox/downloads/file/3773603/watchmarker_for_youtube-4.1.0-fx.xpi

as a +, it syncs history with your yt and browser

**Comment by hdwmp123** [comment]
I have the same problem, it works on mac, but not on ubuntu

mac: 12.5 (21G72)
Docker version 20.10.17, build 100c701
docker-compose version 1.29.2, build 5becea4c

ununtu: #46~20.04.1-Ubuntu SMP Thu Jul 14 15:20:17 UTC 202
Docker version 20.10.17, build 100c701
Docker Compose version v2.3.3

**Pull request merged**